modules/tests/test/modify-vm-template_test.go

```diff
@@ -0,0 +1,209 @@
+				},
+				TaskData: testconfigs.ModifyTemplateTaskData{
+					TemplateName: testtemplate.CirrosTemplateName,
+					CPUCores:     "nonsense value",
+				},
+			}),
+			table.Entry("wrong number of CPU Sockets value provided", &testconfigs.ModifyTemplateTestConfig{
+				TaskRunTestConfig: testconfigs.TaskRunTestConfig{
+					ServiceAccount: ModifyTemplateServiceAccountName,
+					ExpectedLogs:   "parsing \"nonsense value\": invalid syntax",
+				},
+				TaskData: testconfigs.ModifyTemplateTaskData{
+					TemplateName: testtemplate.CirrosTemplateName,
+					CPUSockets:   "nonsense value",
+				},
+			}),
+			table.Entry("wrong number of CPU Threads value provided", &testconfigs.ModifyTemplateTestConfig{
+				TaskRunTestConfig: testconfigs.TaskRunTestConfig{
+					ServiceAccount: ModifyTemplateServiceAccountName,
+					ExpectedLogs:   "parsing \"nonsense value\": invalid syntax",
+				},
+				TaskData: testconfigs.ModifyTemplateTaskData{
+					TemplateName: testtemplate.CirrosTemplateName,
+					CPUThreads:   "nonsense value",
+				},
+			}),
+		)
+	})
+	Context("modify template sucess", func() {
+		table.DescribeTable("taskrun succeded and template is updated", func(config *testconfigs.ModifyTemplateTestConfig, checkResults func(*templatev1.Template)) {
+			f.TestSetup(config)
+
+			if template := config.TaskData.Template; template != nil {
+				t, err := f.TemplateClient.Templates(template.Namespace).Create(context.TODO(), template, v1.CreateOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
+				f.ManageTemplates(t)
+			}
+
+			runner.NewTaskRunRunner(f, config.GetTaskRun()).
+				CreateTaskRun().
+				ExpectSuccess().
+				ExpectLogs(config.GetAllExpectedLogs()...).
+				ExpectResults(map[string]string{
+					"name":      config.TaskData.TemplateName,
+					"namespace": config.TaskData.TemplateNamespace,
+				})
+
+			updatedTemplate, err := f.TemplateClient.Templates(string(config.TaskData.TemplateNamespace)).Get(context.TODO(), config.TaskData.TemplateName, v1.GetOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(updatedTemplate).ToNot(Equal(nil), "new template should exists")
+			f.ManageTemplates(updatedTemplate)
+
+			checkResults(updatedTemplate)
+		},
+			table.Entry("should update template", &testconfigs.ModifyTemplateTestConfig{
+				TaskRunTestConfig: testconfigs.TaskRunTestConfig{
+					ServiceAccount: ModifyTemplateServiceAccountName,
+				},
+				TaskData: testconfigs.ModifyTemplateTaskData{
+					Template:            testtemplate.NewCirrosServerTinyTemplate().Build(),
+					TemplateName:        testtemplate.CirrosTemplateName,
+					CPUCores:            CPUTopologyNumberStr,
+					CPUSockets:          CPUTopologyNumberStr,
+					CPUThreads:          CPUTopologyNumberStr,
+					Memory:              MemoryValue,
+					TemplateLabels:      mockArray,
+					TemplateAnnotations: mockArray,
+					VMAnnotations:       mockArray,
+					VMLabels:            mockArray,
+				},
+			}, func(t *templatev1.Template) {
```

Comment:
can we do this declaratively? ModifyTemplateTaskData can be enhanced with ExpectCPU and so on.. 
This way we can include more positive testcase much more easily

Reply:
even probably better to split it from ModifyTemplateTaskData and have a separate object: something like ExpectedResults and include all the data there